### PR TITLE
Add initial transaction family specification

### PR DESCRIPTION
**Pull request by jsmitchell**
Signed-off-by: James Mitchell <mitchell@bitwise.io>

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'sphinx.ext.todo',
+    'sphinx.ext.ifconfig',
+    'sphinx.ext.autodoc',
+    'sphinx.ext.viewcode',
+    'sphinx.ext.napoleon',
+    'sphinxcontrib.httpdomain',
+    'sphinxcontrib.openapi',
```
Comment (peterschwarz) [inline docs/source/conf.py]: This extension is probably unnecessary, and I would also assume that the `httpdomain` on is not needed either.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

**Review by peterschwarz — APPROVED** [review]

File: docs/source/transaction_family_specification.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+- The first 6 characters of the address are the first 6 hexdigest characters of a sha512 hash of
+  the utf-8 formatted string 'OMI'
+- The next 2 characters of the address are based on the object type:
+
+  - '00' - IndividualIdentity
+  - '01' - OrganizationalIdentity
+  - 'a0' - Work
+  - 'a1' - Recording
+
+- The remaining 62 characters of the address are the last 62 hexdigest characters of a sha512 hash
```
Comment (nick-drozd) [inline docs/source/transaction_family_specification.rst]: It isn't obvious at a glance what an object's unique key is. There is no 'unique key field' -- it's 'name' for identities and 'title' for works and recordings, but you wouldn't know that unless you read through the comments. Is there a way we can make this more obvious?

File: docs/source/transaction_family_specification.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+Summer Lab namespace prefix, the object type prefix, and the unique key field of the object.
+
+Addresses will adhere to the following format:
+
+- Addresses must be a 70 character hexadecimal string
+- The first 6 characters of the address are the first 6 hexdigest characters of a sha512 hash of
+  the utf-8 formatted string 'OMI'
+- The next 2 characters of the address are based on the object type:
+
+  - '00' - IndividualIdentity
```
Comment (nick-drozd) [inline docs/source/transaction_family_specification.rst]: What's the reasoning for these character choices? Can we make them mnemonically easier?

**Review by nick-drozd — CHANGES_REQUESTED** [review]

File: docs/source/transaction_family_specification.rst (hunk 3 above)
Comment (jsmitchell) [reply docs/source/transaction_family_specification.rst]: Can you give me an example of what would be easier?

File: docs/source/transaction_family_specification.rst (hunk 2 above)
Comment (jsmitchell) [reply docs/source/transaction_family_specification.rst]: I'll add some supporting text.

File: docs/source/transaction_family_specification.rst (hunk 3 above)
Comment (nick-drozd) [reply docs/source/transaction_family_specification.rst]: How about the last two characters of the hash of the object type?

```
Work: db
Recording: 2e
IndividualIdentity: 29
OrganizationalIdentity: 2c
```

Comment (jsmitchell) [reply docs/source/transaction_family_specification.rst]: That's easier to remember? :)

Comment (nick-drozd) [reply docs/source/transaction_family_specification.rst]: Well not the characters themselves, but I can always do 
```
>>> import hashlib
>>> f = lambda x: hashlib.sha512(x.encode('utf-8')).hexdigest()[-2:]
>>> f('Work')
'db'
>>> f('Recording')
'2e'
>>> f('IndividualIdentity')
'29'
>>> f('OrganizationalIdentity')
'2c'
```

Comment (jsmitchell) [reply docs/source/transaction_family_specification.rst]:
```
>>> d = {'IndividualIdentity': '00', 'OrganizationalIdentity': '01', 'Work': 'a0', 'Recording': 'a1'}
>>> d['Work']
'a0'
```

**Pull request closed without merging**